### PR TITLE
Adding functionality to Edit, Delete, and Clone security group for Azure

### PR DESCRIPTION
**Pull request by rguthriemsft**
This pull request is to get edit, clone and delete working with Azure.  We are planning to do a refactor of the javascript code after this commit along with move to v2 wizards to bring this functionality in line with current security group UX in spinnaker.

### CODE REVIEW

File: app/scripts/modules/azure/securityGroup/clone/cloneSecurityGroup.controller.js [hunk 1]
```diff
@@ -6,68 +6,154 @@ module.exports = angular
...
+      if (!$state.includes('**.securityGroupDetails')) {
+        $state.go('.securityGroupDetails', newStateParams);
+      } else {
+        $state.go('^.securityGroupDetails', newStateParams);
+      }
+    }
+
+    function onTaskComplete() {
+      application.refreshImmediately();
+      application.registerOneTimeRefreshHandler(onApplicationRefresh);
```
Comment (anotherchrisberry) [inline app/scripts/modules/azure/securityGroup/clone/cloneSecurityGroup.controller.js]: These two calls should probably be replaced with:
```javascript
application.securityGroups.refresh();
```
(sorry, that change just went in this morning)

File: app/scripts/modules/azure/securityGroup/configure/EditSecurityGroupCtrl.js [hunk 2]
```diff
@@ -77,33 +52,105 @@ module.exports = angular.module('spinnaker.azure.securityGroup.azure.edit.contro
...
+      if (!$state.includes('**.securityGroupDetails')) {
+        $state.go('.securityGroupDetails', newStateParams);
+      } else {
+        $state.go('^.securityGroupDetails', newStateParams);
+      }
+    }
+
+    function onTaskComplete() {
+      application.refreshImmediately();
+      application.registerOneTimeRefreshHandler(onApplicationRefresh);
```
Comment (anotherchrisberry) [inline app/scripts/modules/azure/securityGroup/configure/EditSecurityGroupCtrl.js]: same comment as above

**Comment by anotherchrisberry** [comment]
With the exception of the refresh comments above, this looks good - will also need a rebase.

**Comment by rguthriemsft** [comment]
I will do the rebate this morning.  If it's OK, I will defer the task completion change to next commit so I can properly test everything is working ok and bundle with the wizard updates.

**Comment by anotherchrisberry** [comment]
Sure, no problem - let me know when you get it rebased and I'll get this merged asap

**Pull request closed without merging**